src/scenes/ProgressReports/EditForm/Steps/SubmitReportStep/ConfirmVariantDialog.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Typography,
+} from '@mui/material';
+import { Role } from '~/api/schema/schema';
+import { PromptResponseListFragment } from '~/common/fragments';
+import { useSession } from '~/components/Session';
+import { ProgressReportEditFragment } from '../../ProgressReportEdit.graphql';
+
+interface ConfirmEmptyVariantsProps {
+  report: ProgressReportEditFragment;
+  open: boolean;
+  onContinue: () => void;
+  onCancel: () => void;
+  onUpdateConfirmStatus: () => void;
+}
+
+export const ConfirmEmptyVariants = ({
+  report,
+  open,
+  onContinue,
+  onCancel,
+  onUpdateConfirmStatus,
+}: ConfirmEmptyVariantsProps) => {
+  const { reportFile } = report;
+  const { session } = useSession();
+  const roles = new Set(session?.roles.value);
+  const hasReportProgressFile = !!reportFile.value;
+  const projectMgmtHeader = !hasReportProgressFile && reportFile.canEdit;
+
+  const handleConfirmDialog = () => {
+    onContinue();
+  };
+
+  const handleCloseDialog = () => {
+    onCancel();
+  };
+
+  const getEmptyResponsesForRole = (
+    list: PromptResponseListFragment,
+    roles: Set<Role>
+  ) => {
+    if (list.items.length === 0) return [];
+    const emptyResponsesByRole = list.items[0]?.responses.filter((item) => {
+      return item.variant.responsibleRole
+        ? roles.has(item.variant.responsibleRole) &&
+            !item.response.value &&
+            item.response.canEdit
+        : false;
+    });
+    return emptyResponsesByRole ? emptyResponsesByRole : [];
+  };
+
+  const listItemsData = [
+    {
+      key: 'teamNews',
+      condition:
+        !report.teamNews.items.length ||
+        getEmptyResponsesForRole(report.teamNews, roles).length,
+      label: 'Team News',
+    },
+    {
+      key: 'story',
+      condition:
+        !report.communityStories.items.length ||
+        getEmptyResponsesForRole(report.communityStories, roles).length,
+      label: 'Story',
+    },
+  ];
+  const listItems = listItemsData
+    .filter((item) => item.condition)
+    .map((item) => <li key={item.key}>{item.label}</li>);
+
+  const investorConnectionHeader = listItems.length > 0;
+  if (!investorConnectionHeader && !projectMgmtHeader && !open) {
+    onUpdateConfirmStatus();
+  }
```

Comment:
Obviously all of this goes away based on the first comment. But if it hadn't this would be bad. Renders need to be pure, and calling this function on render is a side effect. For this specific case, there's nothing bad, but it's still a smell. The refactor I suggested solves this.